integration_tests/redis-sink/create_sink.sql

```diff
@@ -3,19 +3,13 @@ FROM
     bhv_mv WITH (
     primary_key = 'user_id',
     connector = 'redis',
-    type = 'append-only',
-    force_append_only='true',
     redis.url= 'redis://127.0.0.1:6379/',
-);
+)FORMAT PLAIN ENCODE JSON(force_append_only='true');
 
 CREATE SINK bhv_redis_sink_2
 FROM
     bhv_mv WITH (
     primary_key = 'user_id',
     connector = 'redis',
-    type = 'append-only',
-    force_append_only='true',
     redis.url= 'redis://127.0.0.1:6379/',
-    redis.keyformat='user_id:{user_id}',
-    redis.valueformat='username:{username},event_timestamp{event_timestamp}'
-);
\ No newline at end of file
+)FORMAT PLAIN ENCODE TEMPLATE(force_append_only='true', redis_key_format = 'UserID:{user_id}', redis_value_format = 'TargetID:{target_id},EventTimestamp{event_timestamp}');
```

Comment:
May not add `force_append_only='true'` here?

And may not have the `redis_` prefix for both key_format and value_format.

Reply:
tried it, but without it, success cannot be achieved

Reply:
I mean, for the SQL syntax, we should put the `force_append_only` in the `WITH` part of the SQL rather than in the properties of encoder.

Reply:
cc@xiangjinwu

Reply:
> I mean, for the SQL syntax, we should put the `force_append_only` in the `WITH` part of the SQL rather than in the properties of encoder.

cc @xiangjinwu

Reply:
It is part of `format plain`.

Reply:
`force_append_only` is derived in frontend and stored as `SinkType` separately in SinkDesc, and before the formatter, the sink executor will remove the delete and upsert delete when `force_append_only` is true. And in the [code](https://github.com/risingwavelabs/risingwave/blob/6939da61d9669f19d6b1193cc5eb53780b41a4b9/src/frontend/src/optimizer/plan_node/stream_sink.rs#L213) of frontend, `force_append_only` is derived only by the with options. Should we fix this? @xiangjinwu

Reply:
> `force_append_only` is derived only by the with options

Check line 236-245 below. It has been updated.

Regarding `SinkType`, I actually plan to rename it to `AppendOnlyState = [APPEND_ONLY, FORCE_APPEND_ONLY, NOT_APPEND_ONLY]`, given its current actual functionality is not aligned with `type = ` in old syntax / `format` in new syntax. Its name has been confusing.

Reply:
I see. Good to know that.